### ISSUE TITLE
Bluetooth audio

### ISSUE THREAD

**Issue by erhan-** [icedream/denon-prime4#42]
Hey everyone,

so my idea was to get bluetooth working and connect it to a speaker to see if it is possible to output music to bluetooth as well.

I have to unblock the bluetooth device first via rfkill and then pair and connect to the speaker.
Unfortunately it fails to connect as seen below.
Work in progress.

```
# rfkill
ID TYPE      DEVICE      SOFT      HARD
 0 wlan      phy0   unblocked unblocked
 1 bluetooth hci0     blocked unblocked
# rfkill unblock 1
# bluetoothctl
Agent registered
[CHG] Controller CC:4B:73:72:6F:A1 Pairable: yes
[bluetooth]# power on
Changing power on succeeded
[bluetooth]# scan on
Discovery started
[CHG] Controller CC:4B:73:72:6F:A1 Discovering: yes
[NEW] Device 70:54:B4:44:32:1D TOSHIBA TV
[bluetooth]# connect
70:54:B4:44:32:1D  B8:69:C2:82:B6:87
[bluetooth]# connect B8:69:C2:82:B6:87
Attempting to connect to B8:69:C2:82:B6:87
[CHG] Device B8:69:C2:82:B6:87 Connected: yes
[CHG] Device B8:69:C2:82:B6:87 ServicesResolved: yes
Failed to connect: org.bluez.Error.NotAvailable br-connection-profile-unavailable
[CHG] Device B8:69:C2:82:B6:87 ServicesResolved: no
[CHG] Device B8:69:C2:82:B6:87 Connected: no
```

**Comment by icedream** [comment]
Yep, that's the exact same issue I ran into when I tried this as well. In my case I tried to figure out whether Bluetooth audio could be routed from a device *to* the Prime 4 but it seems some critical files it needs for this to work are missing from the firmware or something like that?

**Comment by erhan-** [comment]
Yes, the troubleshooting online also mentions installing different additional packages. Are we able to output any sound at all via alsa? I think at least the alsamixer does show devices but no controls. I am not home for some days to try out again.

**Comment by erhan-** [comment]
By the way. In many of the Denon facebook groups are talking about bluetooth being added in an updated version. Not sure if it is true or just a rumour.

**Comment by davidd1205** [comment]
hello the bluetooth is for connect phone or connect speaker ? 
Denon has a bt for phone with last firmware in beta test perhaps  but no for output

**Comment by erhan-** [comment]
@davidd1205 can you provide a link to that firmware or any info around it?

**Comment by davidd1205** [comment]
Denon backtracked on the Prime 4 and Prime 2 because Wi-Fi and Bluetooth were causing some errors, but they worked on the Bluetooth input with the Prime 4 and released a version that worked. Could we, from this version, reactivate Bluetooth in version 4, for example?

**Comment by icedream** [comment]
Depends on whether something kernel module related was necessary for this. They did switch between Linux kernel versions if I saw this correctly, so if they added kernel modules for this, there's a good chance the changes are incompatible. If it was just firmware or other software stuff it may be possible to port over. Haven't had time to look into it yet as I have to figure out other complexities with the buildroot build system atm.

**Comment by davidd1205** [comment]
Okay, thank you very much for your response. Keep us informed if you have any updates. Thanks for all your work, anyway.

**Comment by icedream** [comment]
I just recently picked up work on this project again and the changelog for 4.1.0 includes enabling Bluetooth audio output.

I think this issue can be closed once we have modified firmware for 4.1.0 and newer then?

**Comment by erhan-** [comment]
Yes I agree. I tested it once and it was working (albeit the expected lags).
